language/move-prover/src/bytecode_translator.rs

```diff
@@ -46,6 +47,45 @@ pub struct ModuleTranslator<'env> {
     stackless_bytecode: Vec<StacklessFunction>,
 }
 
+/// A struct encapsulating information which is threaded through translating the bytecodes of
+/// a single function. This holds information which is relevant across multiple bytecode
+/// instructions, like borrowing information and label offsets.
+struct BytecodeContext<'l> {
+    /// The stackless bytecode.
+    code: &'l StacklessFunction,
+    /// The bytecode offsets which require a label because they are branched to.
+    branching_targets: BTreeSet<CodeOffset>,
+    /// Set of mutual references, represented by local index. Used for debug tracking. Currently,
+    /// after each mutation (either by an instruction or by call to a function with mutable
+    /// parameters), we dump tracking info for all the variables in this set. This is a vast
+    /// over-approximation; however, the execution trace visualizer will remove redundant
+    /// entries, so it is more of a performance concern.
+    mutual_refs: BTreeSet<usize>,
+    /// A map from local indices  to the before borrow index. Every root which is borrowed in this
+    /// function has an entry here. For every BorrowLoc(d) | BorrowGlobal(d), we have an entry
+    /// d -> i and construct variables $before_borrow_i to remember the value before borrowing,
+    /// and $before_borrow_i_ref to remember the reference. We also track aliasing, i.e. if
+    /// we have MoveLoc(d', d) we add d' -> i as well.
+    borrowed_to_before_index: BTreeMap<usize, usize>,
+    /// As determined by lifetime analysis, the set of references which become dead
+    /// at a given bytecode offset.
+    /// TODO: the analysis currently represents references via `LocalIndex == u8`, but this might(?)
+    /// overflow because the stackless bytcode can introduce temporaries larger than u8.
```

Comment:
Yup. At some point we should change everything to be just usize, starting from stackless bytecode generation.